### PR TITLE
Updated stache to use new data plugin.

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**

### CODE REVIEW

**Comment by Blackbaud-SteveBrush** [comment]
**Plugin should:**
- Handle and parse various file names
- Run unit tests

**Comment by Blackbaud-BrandonJones** [comment]
Update the page-anchor component to use AfterContentChecked instead of ngOnInit, so if we add a variable to a page anchor it will still render in the table of contents.

File: src/app/public/src/modules/page-anchor/page-anchor.component.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class StachePageAnchorComponent implements OnInit, StacheNavLink {
     private elementRef: ElementRef,
     private windowRef: StacheWindowRef) { }
 
-  public ngOnInit(): void {
+  public ngAfterContentChecked(): void {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/src/modules/page-anchor/page-anchor.component.ts]: Curious about why this was changed?

Comment (blackbaud-brandonhare) [inline src/app/public/src/modules/page-anchor/page-anchor.component.ts]: This needed to be changed in order to be able to use variables in the page anchor component. With the way it was before using `ngOnInit`, the content would render, but the TOC would be blank. It's a bug that Lindsey filed a while back.

File: src/app/public/src/modules/shared/json-data.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    expect(data.global.productNameLong).toBe('Stache 2');
+  });
+
+  it('should return data from a specific key', () => {
+    const dataService = new StacheJsonDataService();
+    let data = dataService.getByKey('global');
+    expect(data.productNameLong).toBe('Stache 2');
+  });
+
+  it('should return undefined if the key does not exist', () => {
```
Comment (blackbaud-brandonhare) [inline src/app/public/src/modules/shared/json-data.service.spec.ts]: Ah, good catch!

File: src/app/public/src/modules/page-anchor/page-anchor.component.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [inline src/app/public/src/modules/page-anchor/page-anchor.component.ts]: Boyscoutin' it. Awesome!

Comment (Blackbaud-BrandonJones) [inline src/app/public/src/modules/page-anchor/page-anchor.component.ts]: 👍

**Pull request merged**